### PR TITLE
[voicemail] Handle voicemail bytes with when

### PR DESCRIPTION
**Pull request by didier-wg2**

### CODE REVIEW

File: src/main/kotlin/com/wgtwo/example/voicemail/VoicemailDemo.kt [hunk 1]
```diff
@@ -40,11 +40,10 @@ object VoicemailDemo {
         println(tempFile.absoluteFile)
         val outputStream = tempFile.outputStream()
 
-        if (voicemail.bytesCase != BytesCase.WAV) {
-            println("Unexpected format for voicemail.")
-            return
+        when (voicemail.bytesCase) {
+            BytesCase.WAV -> voicemail.wav.writeTo(outputStream)
+            BytesCase.BYTES_NOT_SET, null -> println("No content found in the voicemail response.")
```
Comment (jorunfa) [inline src/main/kotlin/com/wgtwo/example/voicemail/VoicemailDemo.kt]: What happens when you later try to play a file with nothing in it?

**Review by jorunfa — APPROVED** [review]
1 question

File: src/main/kotlin/com/wgtwo/example/voicemail/VoicemailDemo.kt (hunk 1 above)
Comment (didier-wg2) [reply src/main/kotlin/com/wgtwo/example/voicemail/VoicemailDemo.kt]: Good catch, that failed miserably. I updated the code now.

File: src/main/kotlin/com/wgtwo/example/voicemail/VoicemailMain.kt [hunk 2]
```diff
@@ -8,15 +8,21 @@ import com.github.ajalt.clikt.parameters.options.required
...
+
     init {
         subcommands(List, Play, MarkRead, Delete)
     }
 }
 
-object List: CliktCommand() {
-    val msisdn by option("-m", "--msisdn", help = "Msisdn", envvar = "MSISDN").required()
+object List : CliktCommand() {
+    val msisdn by option(
```
Comment (jorunfa) [inline src/main/kotlin/com/wgtwo/example/voicemail/VoicemailMain.kt]: doesn't msisdn mean that the plus is omitted, which is a contradiction to e164? Rename the val to e162Number (and drop the comment)?

Comment (didier-wg2) [reply src/main/kotlin/com/wgtwo/example/voicemail/VoicemailMain.kt]: Done!

Comment (jorunfa) [reply src/main/kotlin/com/wgtwo/example/voicemail/VoicemailMain.kt]: Rename this val too? And the `-m` -> `-e`?

**Review by jorunfa — APPROVED** [review]

**Pull request merged**